src/MongoDB.Driver.GridFS/GridFSForwardOnlyUploadStream.cs

```diff
@@ -308,6 +354,8 @@ private IEnumerable<BsonDocument> CreateWriteBatchChunkDocuments()
 
         protected override void Dispose(bool disposing)
         {
+            CloseIfNotAlreadyClosedFromDispose(disposing);
+
```

Comment:
This is the key line that fixes the bug.

If the application called Close explicitly then CloseIfNotAlreadyClosedFromDispose will end up doing nothing.

Otherwise, this is the hook that ensures that no matter how we reached Dispose the Stream will be closed.